code/WorkInProgress/Ported/policetape.dm

```diff
@@ -214,5 +219,148 @@
+
+/obj/item/taperoll/syndie/preattack(atom/target, mob/user, proximity_flag, click_parameters)
+	if (charges_left & (istype(target, /obj/machinery/door/airlock) || istype(target, /obj/machinery/door/firedoor)))
+		charges_left--
+		if (!(charges_left))
+			to_chat(user, "<span class = 'warning'>There is no tape left.</span>")
+			qdel(src)
+			return
+		to_chat(user, "<span class = 'notice'>There [charges_left > 1 ? "are" : "is"] [charges_left] roll[charges_left > 1 ? "s" : ""] of tape left.</span>")
+	. = ..()
+
+/obj/item/taperoll/syndie/afterattack(var/atom/A, mob/user, proximity_flag)
+	if (!charges_left)
+		to_chat(user, "<span class = 'warning'>There is no tape left.</span>")
+		qdel(src)
+
+/obj/item/taperoll/syndie/attack_self(var/mob/user)
+	if (charges_left)
+		..()
+		if (icon_state == "[icon_base]_start")
+			charges_left--
+			if (!charges_left)
+				to_chat(user, "<span class = 'warning'>There is no tape left.</span>")
+				qdel(src)
+				return
+			to_chat(user, "<span class = 'notice'>There [charges_left > 1 ? "are" : "is"] [charges_left] roll[charges_left > 1 ? "s" : ""] of tape left.</span>")
+	else
+		to_chat(user, "<span class = 'warning'>There is no tape left.</span>")
+		qdel(src)
+
+// -- Syndie police tape : it cuffs people attempting to attack it. It's also unbreakable by simple mobs.
+
+/obj/item/tape/police/syndie/destroy_tape(var/mob/user, var/obj/item/weapon/W)
+	if (istype(W))
+		if (!W.is_sharp() || !(W.force >= 10))
+			to_chat(user, "<span class='warning'>The tape resists your attack!")
+			return FALSE
+		return ..() // We could destroy it
+	else // Attacks with bare hands, cuffs himself on it
+		if (ishuman(user))
+			var/mob/living/carbon/human/H = user
+			if (H.has_organ_for_slot(slot_handcuffed))
+				H.visible_message("<span class='danger'>[H] wraps \his hands on the tape!</span>", "<span class='danger'>The tape wraps itself on your hands!</span>")
+				var/obj/item/taperoll/police/cuffs = new
+				H.equip_to_slot(cuffs, slot_handcuffed)
+				cuffs.on_restraint_apply(H)
+		return FALSE
+
+
+/obj/item/tape/police/syndie/examine(mob/user)
+	. = ..()
+	if (get_dist(user, src) < 3)
+		to_chat(user, "<span class = 'warning'>This one looks heavier than the usual.</span>")
+
+// -- Syndie atmos tape : shocks and sparks you (useful for lighting those plasma fires)
+
+/obj/item/tape/engineering/syndie
+	var/charged = TRUE
+	siemens_coefficient = 1
+
+/obj/item/tape/engineering/syndie/destroy_tape(var/mob/user)
+	if (spark_and_shock(user)) // If you were shocked, you couldn't destroy the tape !
+		return FALSE
+	return ..()
+
+/obj/item/tape/engineering/syndie/proc/spark_and_shock(var/mob/user)
+	if (user && charged)
+		spark(src, 5)
+		return shock(user, 50)
+	else // No user, or not charged
+		return FALSE
+
+/obj/item/tape/engineering/syndie/proc/shock(var/mob/user, var/damage)
+	if (!user || !istype(user, /mob/living))
+		return FALSE
+	if (ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(H.gloves)
```

Comment:
use `get_item_by_slot` instead
call it once and re-use it in the line below this one